pkg/targets/proxy.go

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package targets
+
+import (
+	"context"
+	"io"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/pkg/errors"
+	"github.com/thanos-io/thanos/pkg/store/storepb"
+	"github.com/thanos-io/thanos/pkg/targets/targetspb"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// Proxy implements targetspb.Targets gRPC that fanouts requests to given targetspb.Targets and deduplication on the way.
```

Comment:
Nit:
```suggestion
// Proxy implements targetspb.Targets gRPC that fans out requests to given targetspb.Targets and deduplicates on the way.
```

Reply:
The comment says that Proxy implements deduplication, but the methods have no deduplication code. Instead, it seems like it's the responsibility of the caller to deduplicate. Let's either fix the comment or add the deduplication logic here.